subsys/net/lib/config/init_clock_sntp.c

```diff
@@ -82,40 +84,263 @@ static void sntp_set_rtc(__maybe_unused const struct timespec *tspec)
+{
+	int ret;
+
+	if (status == DNS_EAI_CANCELED || status == DNS_EAI_FAIL) {
+		/* If IPv4 query failed, try IPv6. Otherwise, just schedule next retry. */
+		if (sntp_addr.ss_family == NET_AF_INET && IS_ENABLED(CONFIG_NET_IPV6)) {
+			sntp_addr.ss_family = NET_AF_INET6;
+			sntp_addrlen = 0;
+
+			ret = dns_get_addr_info(CONFIG_NET_CONFIG_SNTP_INIT_SERVER,
+						DNS_QUERY_TYPE_AAAA, NULL, dns_result_cb,
+						NULL, CONFIG_NET_CONFIG_SNTP_INIT_TIMEOUT);
+			if (ret == 0) {
+				return;
+			}
+		}
+
+		LOG_WRN("DNS query timed out");
+
+		k_work_reschedule(&sntp_resync_work_handle, RESYNC_FAILED_INTERVAL);
+
+		return;
+	}
+
+	if (status == DNS_EAI_ALLDONE) {
+		/* If address was found, schedule SNTP query, if that fails schedule next retry. */
+		ret = sntp_query_async(net_sad(&sntp_addr), sntp_addrlen);
+		if (ret < 0) {
+			k_work_reschedule(&sntp_resync_work_handle, RESYNC_FAILED_INTERVAL);
+		}
+
+		return;
+	}
+
+	if (status == DNS_EAI_INPROGRESS && info != NULL) {
+		if (sntp_addrlen > 0) {
+			/* Already got address, skip others. */
+			return;
+		}
+
+		if (IS_ENABLED(CONFIG_NET_IPV4) && info->ai_family == NET_AF_INET) {
+			sntp_addrlen = info->ai_addrlen;
+			sntp_addr.ss_family = NET_AF_INET;
+			net_ipv4_addr_copy_raw(net_sin(net_sad(&sntp_addr))->sin_addr.s4_addr,
+					       net_sin(&info->ai_addr)->sin_addr.s4_addr);
+			net_sin(net_sad(&sntp_addr))->sin_port = net_htons(SNTP_SERVER_PORT);
+		} else if (IS_ENABLED(CONFIG_NET_IPV6) && info->ai_family == NET_AF_INET6) {
+			sntp_addrlen = info->ai_addrlen;
+			sntp_addr.ss_family = NET_AF_INET6;
+			net_ipv6_addr_copy_raw(net_sin6(net_sad(&sntp_addr))->sin6_addr.s6_addr,
+					       net_sin6(&info->ai_addr)->sin6_addr.s6_addr);
+			net_sin6(net_sad(&sntp_addr))->sin6_port = net_htons(SNTP_SERVER_PORT);
+		}
+	}
+}
+
+static int dns_query_async(void)
+{
+	enum dns_query_type type;
+	int ret;
+
+	memset(&sntp_addr, 0, sizeof(sntp_addr));
+	sntp_addrlen = 0;
+
+	if (IS_ENABLED(CONFIG_DNS_RESOLVER)) {
+		if (IS_ENABLED(CONFIG_NET_IPV4)) {
+			sntp_addr.ss_family = NET_AF_INET;
+			type = DNS_QUERY_TYPE_A;
+		} else {
+			sntp_addr.ss_family = NET_AF_INET6;
+			type = DNS_QUERY_TYPE_AAAA;
+		}
+
+		ret = dns_get_addr_info(CONFIG_NET_CONFIG_SNTP_INIT_SERVER,
+					type, NULL, dns_result_cb,
+					NULL, CONFIG_NET_CONFIG_SNTP_INIT_TIMEOUT);
+		if (ret < 0) {
+			LOG_ERR("Failed to initiate DNS query for SNTP server (%d)", ret);
+		}
+
+		return ret;
+	}
+
+	/* Fallback for IP address string. */
+	if (net_ipaddr_parse(CONFIG_NET_CONFIG_SNTP_INIT_SERVER,
+			     sizeof(CONFIG_NET_CONFIG_SNTP_INIT_SERVER) - 1,
+			     net_sad(&sntp_addr))) {
+		if (sntp_addr.ss_family == NET_AF_INET) {
+			sntp_addrlen = sizeof(struct net_sockaddr_in);
+			net_sin(net_sad(&sntp_addr))->sin_port = net_htons(SNTP_SERVER_PORT);
+		} else if (IS_ENABLED(CONFIG_NET_IPV6)) {
+			sntp_addrlen = sizeof(struct net_sockaddr_in6);
+			net_sin6(net_sad(&sntp_addr))->sin6_port = net_htons(SNTP_SERVER_PORT);
+		}
```

Comment:
why not using ``net_port_set_default()`` like in ``sntp_simple_addr()``

Reply:
Totally forgot we had function like this. Good point about not overwriting custom port number, I've updated both cases to use `net_port_set_default()` instead of setting port manually.